### ISSUE TITLE
Add indicator for private-browsing.

### ISSUE THREAD

**Issue by Earnestly** [The-Compiler/qutebrowser#1052]
This may be a more general request where custom elements can be added to the status bar but if that idea is too impractical I think that an indicator for private-browsing is an important special case to add.  A simple `[P]` would be enough on the right side of the statusbar.

**Comment by vyp** [comment]
Good idea, +1. I think for now it would be simpler to just add an indicator specifically for private browsing. Then later if people find that they want other types of indicators, a more generic solution allowing custom statusbar elements could be made. Unless @The-Compiler disagrees of course.

And yes I agree it should be something simple like `[P]` so it doesn't take up too much space.

**Comment by The-Compiler** [comment]
This is definitely needed after there are per-domain settings and a script-blocker as well, with something like `[S]`.

I'd like to find some generic way for this (like making it possible to add an indicator for any bool config option), but I'm struggling to find a good way of configuring this.

**Comment by Earnestly** [comment]
A generic method might be too complex for what it's worth, "When you narrow your scope you can achieve much greater things" -- Carmack.

The only way to really make for useful custom statusbar items is to make the api for adding python scripts to `qutebrowser/mainwindow/statusbar` more straight forward.  "The universal interface is a bytestream [text]" -- McIlroy.

I think a lot of systems can learn a great deal from the crucial discovery made in Plan9 that essentially reduced the complexity by almost a factor 10 or more as they completely gutted the need for pthreads, ioctls, and more.  The discovery is that everything is a filesystem, not a file.

**Comment by craftyguy** [comment]
For what it's worth (to help folks until this is implemented), I added this to my keys.conf to add a visual marker in the tab when private mode is enabled. You can adjust colors, etc. I use AwesomeFont for the "icon"

In the config below, ctrl+P enabled privacy browsing, and ctrl+shift+P disabled it.

```
set -t general private-browsing true;; set -t colors tabs.fg.selected.even '#da2e5c';;set -t colors tabs.fg.selected.odd '#da2e5c' ;; set -t tabs title-format ' {index}: {title}';; message-info "Privacy Mode ENABLED!"
    <ctrl-p>

set -t general private-browsing false;; set -t colors tabs.fg.selected.odd '#657B83';;set -t colors tabs.fg.selected.even '#657B83';; set -t tabs title-format '{index}: {title}';; message-info "Privacy Mode DISABLED!"
    <ctrl-shift-p>
```

**Comment by Earnestly** [comment]
Interesting approach, this is a little inelegant but effortlessly practical

**Comment by merspieler** [comment]
> making it possible to add an indicator for any bool config option

I'd like to have a way, where it's possible for non-bool options, under a condition, an indicator is shown.

eg. if  `content.proxy = <tor proxy>` then show `[T]`

**Comment by xelxebar** [comment]
FWIW, I'm of a similar mind regarding the desire for a generically customizable status. In my case, I frequently need to toggle the variables `content.javascript.enable` and `content.proxy`, and it would be quite nice to have inconspicuous indicators for the state of these variables.  

In fact, we'd be in good company with such a customizable status: *cf* the relevant manual pages of [uzbl][0], [mutt][1], [lemonbar][2], *etc*. I like @Earnestly's circumspection (I'm also a huge fan of Plan 9!), though I don't think that generality/flexibility equates to complexity. Engineered correctly, often the opposite is true.

[0]:https://www.uzbl.org/readme.php
[1]:https://www.mutt.org/doc/manual/#status-format
[2]:https://wiki.archlinux.org/index.php/Lemonbar

**Comment by tamis-laan** [comment]
Same here, in need of this feature.

**Comment by The-Compiler** [comment]
@tamis-laan Note that you can [add a :+1: reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to an issue, which 1) is more visible (and e.g. usable as a sort criteria for issues), and 2) avoids sending notifications to ~180 people watching this repo.
